core/tests/protractor/userJourneys.js

```diff
@@ -342,46 +415,40 @@ describe('Site language', function() {
   });
 
   it('should not change in an exploration', function() {
-    users.createUser('mangue@example.com', 'Mangue');
-    users.login('mangue@example.com', true);
+    users.login('langCreator@explorations.com', true);
     browser.get('/about');
     _selectLanguage('Español');
 
-    // Create an exploration.
-    workflow.createExploration();
-    general.getExplorationIdFromEditor().then(function(expId) {
-      explorationId = expId;
-      editor.setContent(forms.toRichText('Language Test'));
-      editor.setInteraction('NumericInput');
-      editor.addResponse(
-        'NumericInput', forms.toRichText('Nice!!'),
-        'END', true, 'IsLessThanOrEqualTo', 0);
-      editor.setDefaultOutcome(forms.toRichText('Ok!!'), null, false);
-      editor.moveToState('END');
-      editor.setContent(forms.toRichText('END'));
-      editor.setInteraction('EndExploration');
+    general.openEditor(firstExplorationId);
 
-      // Save changes.
-      title = 'Language Test';
-      category = 'Languages';
-      objective = 'To test site language.';
-      editor.setTitle(title);
-      editor.setCategory(category);
-      editor.setObjective(objective);
-      editor.saveChanges('Done!');
+    // Spanish is still selected.
+    var placeholder = element(by.css('.protractor-test-float-form-input'))
+      .getAttribute('placeholder');
+    expect(placeholder).toEqual('Ingresa un número');
+    general.ensurePageHasNoTranslationIds();
+    users.logout();
+  });
 
-      // Publish changes.
-      workflow.publishExploration();
-      general.openEditor(expId);
+  it('should not change in exploration and collection player for guest users',
+    function() {
+      browser.get('/about');
+      _selectLanguage('Español');
 
-      // Spanish is still selected.
-      var placeholder = element(by.css('.protractor-test-float-form-input'))
-        .getAttribute('placeholder');
-      expect(placeholder).toEqual('Ingresa un número');
+      // Checking collection player page.
+      browser.get('/collection/' + collectionId);
+      browser.waitForAngular();
+      expect(element(by.css('.oppia-share-collection-footer')).getText())
+        .toEqual('COMPARTIR ESTA COLECCIÓN');
```

Comment:
@U8NWXD Does not the code and it block contradicts each other. The it blocks says that site language should not change in exploration and collection player for guest users. But here, we are checking the text to be in spanish.